src/main/java/com/salesforce/dynamodbv2/mt/mappers/MtAmazonDynamoDbComposite.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (c) 2019, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root  or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+package com.salesforce.dynamodbv2.mt.mappers;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.amazonaws.services.dynamodbv2.model.BatchGetItemRequest;
+import com.amazonaws.services.dynamodbv2.model.BatchGetItemResult;
+import com.amazonaws.services.dynamodbv2.model.CreateTableRequest;
+import com.amazonaws.services.dynamodbv2.model.CreateTableResult;
+import com.amazonaws.services.dynamodbv2.model.DeleteItemRequest;
+import com.amazonaws.services.dynamodbv2.model.DeleteItemResult;
+import com.amazonaws.services.dynamodbv2.model.DeleteTableRequest;
+import com.amazonaws.services.dynamodbv2.model.DeleteTableResult;
+import com.amazonaws.services.dynamodbv2.model.DescribeTableRequest;
+import com.amazonaws.services.dynamodbv2.model.DescribeTableResult;
+import com.amazonaws.services.dynamodbv2.model.GetItemRequest;
+import com.amazonaws.services.dynamodbv2.model.GetItemResult;
+import com.amazonaws.services.dynamodbv2.model.ListTablesResult;
+import com.amazonaws.services.dynamodbv2.model.PutItemRequest;
+import com.amazonaws.services.dynamodbv2.model.PutItemResult;
+import com.amazonaws.services.dynamodbv2.model.QueryRequest;
+import com.amazonaws.services.dynamodbv2.model.QueryResult;
+import com.amazonaws.services.dynamodbv2.model.ScanRequest;
+import com.amazonaws.services.dynamodbv2.model.ScanResult;
+import com.amazonaws.services.dynamodbv2.model.UpdateItemRequest;
+import com.amazonaws.services.dynamodbv2.model.UpdateItemResult;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.Objects;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+/**
+ * An {@link MtAmazonDynamoDb} that determines which of multiple {@link MtAmazonDynamoDb}s to delegate to based on
+ * context.
+ *
+ * <p>TODO: support backup-related functionality
+ */
+public class MtAmazonDynamoDbComposite extends MtAmazonDynamoDbBase {
+
+    private final Collection<MtAmazonDynamoDbBase> delegates;
+    private final Supplier<MtAmazonDynamoDbBase> fromContextToDelegate;
+    private final Function<String, MtAmazonDynamoDbBase> fromPhysicalTableNameToDelegate;
+
+    public MtAmazonDynamoDbComposite(Collection<MtAmazonDynamoDbBase> delegates,
+                                     Supplier<MtAmazonDynamoDbBase> fromContextToDelegate,
+                                     Function<String, MtAmazonDynamoDbBase> fromPhysicalTableNameToDelegate) {
```

Comment:
Basically the only difference from the previous PR is here, that instead of explicitly passing in the parent MtAmazonDynamoDbBase arguments like the mt context provider, validate all delegates share the same such arguments, and then use the first delegate's properties.